### PR TITLE
Make letter layout primary key a UUID.

### PR DESCRIPTION
**Pull request by craiga**
Note that this deletes all existing migrations in `letter_layouts` and `letter_templates`.

Changing primary keys while maintaining existing migrations is very, very difficult (maybe even impossible).

### CODE REVIEW

File: static/letter_layouts/serializers.py [hunk 1]
```diff
@@ -6,4 +6,4 @@
 class LetterLayoutSerializer(serializers.ModelSerializer):
     class Meta:
         model = LetterLayout
-        fields = '__all__'
```
Comment (craiga) [inline static/letter_layouts/serializers.py]: I'd argue that using `__all__` is bad practice.

We might add an extra fields to a model down the road which contains sensitive information. If we use `__all__` on serializers like this, that data might accidentally be leaved via the API.

**Comment by craiga** [comment]
FYI, before switching to this branch, be sure to run the following:

```bash
pipenv run ./manage.py migrate letter_templates zero
pipenv run ./manage.py migrate letter_layouts zero
```

File: static/letter_layouts/models.py [hunk 2]
```diff
@@ -1,9 +1,12 @@
+import uuid
+
 from django.db import models
 
 
 class LetterLayout(models.Model):
-    id = models.CharField(primary_key=True, editable=False, max_length=30)  # letter file name minus extension
-    name = models.CharField(max_length=30)  # friendly name
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    name = models.TextField(help_text="Friendly name")
```
Comment (BenAndrew-Kainos) [inline static/letter_layouts/models.py]: What is `"Friendly name"`

File: static/letter_layouts/serializers.py (hunk 1 above)
Comment (BenAndrew-Kainos) [reply static/letter_layouts/serializers.py]: That is quite a good point (it's a bit like using `SELECT * FROM ...`). Something to raise with the devs tomorrow I'd say

File: static/letter_layouts/models.py (hunk 2 above)
Comment (craiga) [reply static/letter_layouts/models.py]: The name that's shown in the UI.

I was just moving the comment into [`help_text`](https://docs.djangoproject.com/en/2.2/ref/models/fields/#help-text) as it seemed like a more useful place for it.

Comment (BenAndrew-Kainos) [reply static/letter_layouts/models.py]: Ok can we move this to lite-content please

**Comment by BenAndrew-Kainos** [comment]
> FYI, before switching to this branch, be sure to run the following:
> 
> ```shell
> pipenv run ./manage.py migrate letter_templates zero
> pipenv run ./manage.py migrate letter_layouts zero
> ```

Why does it include the word zero?

**Pull request merged**